inprocess/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -786,6 +790,28 @@ public void writeMessage(InputStream message) {
           statsTraceCtx.outboundMessageSent(outboundSeqNo, -1, -1);
           serverStream.statsTraceCtx.inboundMessage(outboundSeqNo);
           serverStream.statsTraceCtx.inboundMessageRead(outboundSeqNo, -1, -1);
+
+          long messageLength = -1;
+          try {
+            messageLength = getKnownLength(message);
+          } catch (IOException e) {
+            throw Status.INTERNAL
+                    .withDescription("Failed to calculate size")
+                    .withCause(e)
+                    .asRuntimeException();
+          } catch (StatusRuntimeException e) {
+            throw e;
```

Comment:
Why do we need to treat this exception different? Why not have a single catch statement for all 3 exception types?

Reply:
Motivation is from MessageFramer, [here](https://github.com/grpc/grpc-java/blob/9bed655c568b4f09a32b3910745a949f5f08d956/core/src/main/java/io/grpc/internal/MessageFramer.java#L138)